hypermap/aggregator/solr.py

```diff
@@ -41,19 +44,79 @@ def layer_to_solr(self, layer):
         layer_dict = layer2dict(layer)
         layer_json = json.dumps(layer_dict)
         try:
-            url_solr_update = '%s/update/json/docs' % settings.SEARCH_URL
+            url_solr_update = '%s/solr/hypermap/update/json/docs' % SEARCH_URL
             headers = {"content-type": "application/json"}
             params = {"commitWithin": 1500}
-            requests.post(url_solr_update, data=layer_json, params=params,  headers=headers)
-            logger.info("Solr record saved for layer with id: %s" % layer.id)
+            res = requests.post(url_solr_update, data=layer_json, params=params,  headers=headers)
+            res = res.json()
+            if 'error' in res:
+                message = res["error"].get("msg")
+                LOGGER.debug("ERROR: {0}".format(message))
+                return False, message
+            else:
+                LOGGER.info("Solr record saved for layer with id: %s" % layer.id)
+
             return True, None
-        except Exception:
-            logger.error("Error saving solr record for layer with id: %s - %s" % (layer.id, sys.exc_info()[1]))
+        except Exception, e:
+            LOGGER.error("Error saving solr record for layer with id: %s" % layer.id)
+            LOGGER.error(e, exc_info=True)
             return False, sys.exc_info()[1]
 
-    def clear_solr(self):
+    def clear_solr(self, catalog="hypermap"):
         """Clear all indexes in the solr core"""
-        solr_url = settings.SEARCH_URL
+        solr_url = "{0}/solr/{1}".format(SEARCH_URL, catalog)
         solr = pysolr.Solr(solr_url, timeout=60)
         solr.delete(q='*:*')
-        print 'Solr core cleared'
+        LOGGER.debug('Solr core cleared')
+
+    def update_schema(self, catalog="hypermap"):
+        """
```

Comment:
Excellent. We need a timestamp field, I think this is useful for you as well:
<field name="timestamp" type="date" indexed="true" stored="true" default="NOW" multiValued="false"/>